src/source/PeerConnection/SessionDescription.c

```diff
@@ -214,32 +217,42 @@ STATUS setPayloadTypesFromOffer(PHashTable codecTable, PHashTable rtxTable, PSes
                 CHK_STATUS(hashTableUpsert(codecTable, RTC_CODEC_ALAW, parsedPayloadType));
             }
 
-            if ((end = STRSTR(attributeValue, RTX_VALUE)) != NULL) {
-                CHK_STATUS(STRTOUI64(attributeValue, end - 1, 10, &rtxPayloadType));
-                attributeValue = pMediaDescription->sdpAttributes[++currentAttribute].attributeValue;
-                if ((end = STRSTR(attributeValue, RTX_CODEC_VALUE)) != NULL) {
-                    CHK_STATUS(STRTOUI64(end + STRLEN(RTX_CODEC_VALUE), NULL, 10, &parsedPayloadType));
-                    CHK_STATUS(
-                        hashTableContains(codecTable, RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE, &supportCodec));
-                    if (supportCodec) {
-                        CHK_STATUS(
-                            hashTableGet(codecTable, RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE, &hashmapPayloadType));
-                        if (parsedPayloadType == hashmapPayloadType) {
-                            CHK_STATUS(hashTableUpsert(rtxTable, RTC_RTX_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE,
-                                                       rtxPayloadType));
-                        }
-                    }
+            if ((end = STRSTR(attributeValue, RTX_CODEC_VALUE)) != NULL) {
+                CHK_STATUS(STRTOUI64(end + STRLEN(RTX_CODEC_VALUE), NULL, 10, &parsedPayloadType));
+                if ((end = STRSTR(attributeValue, FMTP_VALUE)) != NULL) {
+                    CHK_STATUS(STRTOUI64(end + STRLEN(FMTP_VALUE), NULL, 10, &fmtpVal));
+                    singleListInsertItemHead(pListAptFmtpVals, (fmtpVal << 8u) & parsedPayloadType);
+                }
+            }
+        }
 
-                    CHK_STATUS(hashTableContains(codecTable, RTC_CODEC_VP8, &supportCodec));
-                    if (supportCodec) {
-                        CHK_STATUS(hashTableGet(codecTable, RTC_CODEC_VP8, &hashmapPayloadType));
-                        if (parsedPayloadType == hashmapPayloadType) {
-                            CHK_STATUS(hashTableUpsert(rtxTable, RTC_RTX_CODEC_VP8, rtxPayloadType));
-                        }
-                    }
+        singleListGetNodeCount(pListAptFmtpVals, &nodeCount);
+
+        for (i = 0; i < nodeCount; i++) {
+            singleListGetNodeDataAt(pListAptFmtpVals, i, &aptFmtVal);
```

Comment:
Just a note that this will make your algorithm quadratic as getting node at N is a linear op. If you decide to go with the list solution then you should perhaps use an iterator here to linearly scan the list.

node = singleListGetHead();
while (node != NULL) {
...
node = singleListGetNextNode();
}

something like this - don't remember the syntax.

Reply:
Yeah I wasn't concerned with performance here since as you mentioned there aren't many of these, I would just pick from the head, delete and move on -- however I'm going to go with the fixed size array approach so this wouldn't be an issue.